be/src/storage/tablet_updates.cpp

```diff
@@ -1606,12 +1606,16 @@ void TabletUpdates::_erase_expired_versions(int64_t expire_time,
         LOG(WARNING) << "tablet deleted when erase_expired_versions tablet:" << _tablet.tablet_id();
         return;
     }
-    for (int i = 0; i < _apply_version_idx; i++) {
```

Comment:
just iterate to `_apply_version_idx - 1` also solve this problem?

Reply:
it solves this problem, but introduces other problems, for example: if we write at day1, then write again at day2, and never write again, those 2 versions will live forever cause GC cannot remove them, the correct behavior should only keep 1 version only when data are stable.

Reply:
think about it again, it only solve the case that one new version is published, it doesn't solve the case that there are multiple publish tasks finished in a short time.